src/vaadin-grid-drag-and-drop-mixin.html

```diff
@@ -72,37 +72,49 @@
         if (this.rowsDraggable) {
           let row = e.target;
           if (row.localName === 'vaadin-grid-cell-content') {
-            // In browsers that support shadow, the draggable node is the cell content element
+            // The draggable node is the cell content element on browsers that support native shadow
             row = row.assignedSlot.parentNode.parentNode;
           }
-          if (row.localName !== 'tr') {
+          if (row.parentNode !== this.$.items) {
             return;
           }
 
           e.stopPropagation();
           this._toggleAttribute('dragging-rows', true, this);
 
+          const rowRect = row.getBoundingClientRect();
           if (!window.ShadyDOM) {
-            // Safari doesn't get proper drag images from transformed
-            // elements so we need to switch to top temporarily
-            const transform = row.style.transform;
-            row.style.top = /translateY\((.*)\)/.exec(transform)[1];
-            row.style.transform = 'none';
-
-            const rect = row.getBoundingClientRect();
-            e.dataTransfer.setDragImage(row, e.clientX - rect.left, e.clientY - rect.top);
-
-            requestAnimationFrame(() => {
-              row.style.top = '';
-              row.style.transform = transform;
-            });
+            if (this._safari) {
+              // Safari doesn't get proper drag images from transformed
+              // elements so we need to switch to top temporarily
+              const transform = row.style.transform;
+              row.style.top = /translateY\((.*)\)/.exec(transform)[1];
+              row.style.transform = 'none';
+              requestAnimationFrame(() => {
+                row.style.top = '';
+                row.style.transform = transform;
+              });
+            }
+
+            e.dataTransfer.setDragImage(row, e.clientX - rowRect.left, e.clientY - rowRect.top);
           }
 
           const items = this._isSelected(row._item) ? this.selectedItems.slice(0) : [row._item];
           e.dataTransfer.setData('text', this.formatDragData(items));
 
-          row.setAttribute('dragged-row-count', items.length);
-          requestAnimationFrame(() => row.removeAttribute('dragged-row-count'));
+          row.setAttribute('dragstart', '');
```

Comment:
Was the naming already agreed? Sounds a little bit confusing to me. Not a blocker.

Reply:
What sounds confusing about this name? Any other suggestions?